client/watcher/watcher.go

```diff
@@ -71,29 +72,43 @@ func (w *Watcher) Init() error {
 func (w *Watcher) Watch() {
 	done := make(chan bool)
 	go func() {
+		defer close(done)
+
 		for {
 			w.mu.Lock()
 			select {
-			case event := <-w.watcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write ||
-					event.Op&fsnotify.Create == fsnotify.Create ||
-					event.Op&fsnotify.Chmod == fsnotify.Chmod {
+			case event, ok := <-w.watcher.Events:
+				if !ok {
+					w.mu.Unlock()
```

Comment:
idea: if we put the entire contents of the main `for{}` into its own function, even an anonymous one, then we can use `defer` for the unlock and don't have to make sure we catch all the exit cases

I'm also a little confused about this lock, since it seems we hold it most all the time... and we just have one `Watcher` for the whole app, yes? Is the lock just to protect us when we notice two events in quick succession? Maybe there is a more lightweight locking strategy... not sure.

Reply:
This was code that was already in place (other than the new `Unlock` call), so I hesitate to change it too much. I wasn't sure about the reasoning for the mutex surrounding the `select` either, but assumed it was for good reason?